org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/SynchronizeGradleBuildOperation.java

```diff
@@ -189,17 +183,27 @@ private void synchronizeGradleProjectWithWorkspaceProject(OmniEclipseProject pro
 
     private void synchronizeWorkspaceProject(OmniEclipseProject project, IProject workspaceProject, SubMonitor progress) throws CoreException {
         if (workspaceProject.isAccessible()) {
-            synchronizeOpenWorkspaceProject(project, workspaceProject, progress);
+            synchronizeOpenWorkspaceProject(project, workspaceProject, true, progress);
         } else {
             synchronizeClosedWorkspaceProject(progress);
         }
     }
 
-    private void synchronizeOpenWorkspaceProject(OmniEclipseProject project, IProject workspaceProject, SubMonitor progress) throws CoreException {
+    private void synchronizeOpenWorkspaceProject(OmniEclipseProject project, IProject workspaceProject, boolean refreshNeeded, SubMonitor progress) throws CoreException {
         progress.setWorkRemaining(10);
 
         //currently lots of our synchronization logic assumes that the whole resource tree is readable.
-        CorePlugin.workspaceOperations().refreshProject(workspaceProject, progress.newChild(1));
+        if (refreshNeeded) {
```

Comment:
⭕️ Why not always do a refresh here and leave the workspace operations unchanged? Does the synchronous call not replace the background call?

Reply:
I don't think it does but I'll check just to be sure.

Reply:
I've checked again, the synchronous call doesn't replace the background call. Also, the background job starts with a 200-millisecond delay, so it's indeterministic if the background refresh finishes before or after Gradle project synchronization.